packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -56,9 +95,15 @@ export interface NuxtLinkProps extends Omit<RouterLinkProps, 'to'> {
    */
   prefetchedClass?: string
   /**
-   * When enabled will prefetch middleware, layouts and payloads of links in the viewport.
+   * When enabled, will prefetch middleware, layouts and payloads of links in the viewport.
    */
   prefetch?: boolean
+  /**
+   * Determines when to prefetch links. Defaults to `on-visible`. See
+   * `prefetchMode` for more details. This prop is only used when `prefetch` is
+   * enabled and `noPrefetch` is not set.
+   */
+  prefetchMode?: 'on-hover' | 'on-visible'
```

Comment:
@danielroe I think adding a prop such as `prefetchMode` makes more sense than changing the current `prefetch` to a union type with diversification. What do you think about this?

Reply:
That does make sense I think, as we could also support setting prefetch mode globally or even creating a custom Nuxt link component.